CreeDictionary/API/search/presentation.py

```diff
@@ -0,0 +1,177 @@
+            ret["wordform_text"] = self.wordform
+        else:
+            ret["wordform_text"] = self.wordform.text
+
+        return ret
+
+    @property
+    def relevant_tags(self) -> Tuple[LinguisticTag, ...]:
+        """
+        Tags and features to display in the linguistic breakdown pop-up.
+        This omits preverbs and other features displayed elsewhere
+
+        In itwêwina, these tags are derived from the suffix features exclusively.
+        We chunk based on the English relabelleings!
+        """
+        return tuple(
+            linguistic_tag_from_fst_tags(fst_tags)
+            for fst_tags in LABELS.english.chunk(self.linguistic_breakdown_tail)
+        )
+
+    def __str__(self):
+        return f"PresentationResult<{self.wordform}:{self.wordform.id}>"
+
+
+def serialize_wordform(wordform) -> SerializedWordform:
+    """
+    Intended to be passed in a JSON API or into templates.
+
+    :return: json parsable result
+    """
+    result = model_to_dict(wordform)
+    result["definitions"] = serialize_definitions(wordform.definitions.all())
+    result["lemma_url"] = wordform.get_absolute_url()
+
+    # Displayed in the word class/inflection help:
+    result["inflectional_category_plain_english"] = LABELS.english.get(
+        wordform.inflectional_category
+    )
+    result["inflectional_category_linguistic"] = LABELS.linguistic_long.get(
+        wordform.inflectional_category
+    )
+    result["wordclass_emoji"] = wordform.get_emoji_for_cree_wordclass()
+    result["wordclass"] = wordform.wordclass_text
+
+    return result
+
+
+def serialize_definitions(definitions, include_auto_definitions=False):
+    ret = []
+    for definition in definitions:
+        serialized = definition.serialize()
+        if include_auto_definitions or "auto" not in serialized["source_ids"]:
+            ret.append(serialized)
+    return ret
+
+
+def safe_partition_analysis(analysis: ConcatAnalysis):
+    try:
+        (
+            linguistic_breakdown_head,
+            _,
+            linguistic_breakdown_tail,
+        ) = partition_analysis(analysis)
+    except ValueError:
+        linguistic_breakdown_head = []
+        linguistic_breakdown_tail = []
+    return linguistic_breakdown_head, linguistic_breakdown_tail
+
+
+def replace_user_friendly_tags(fst_tags: List[FSTTag]) -> List[Label]:
+    """ replace fst-tags to cute ones"""
+    return LABELS.english.get_full_relabelling(fst_tags)
+
+
+def get_preverbs_from_head_breakdown(
+    head_breakdown: List[FSTTag],
+) -> Tuple[Preverb, ...]:
+    results = []
+
+    for tag in head_breakdown:
+        preverb_result: Optional[Preverb] = None
+        if tag.startswith("PV/"):
+            # use altlabel.tsv to figure out the preverb
+
+            # ling_short looks like: "Preverb: âpihci-"
+            ling_short = LABELS.linguistic_short.get(tag)
+            if ling_short is not None and ling_short != "":
+                # looks like: "âpihci"
```

Comment:
This comment seems to disagree with the one above. How does it go from "Preverb: âpihci-" to "âpihci"?

Reply:
The line below the comment transforms `Preverb: âpihci-` to `âpihci`; I have updated the comment to try to clarify that.

Reply:
Ohh, I see now.